bazel/external_crates.bzl

```diff
@@ -765,7 +786,6 @@ def external_crates_repository(name, cargo_lockfile, lockfile, sanitizers_enable
             ),
             "libfuzzer-sys": crate.spec(
                 version = "^0.4.7",
-                default_features = False,
```

Comment:
Why do we need to remove this? This would break AFL. we have to disable `default_features`, specifically `link_libfuzzer` since we want to use libfuzzer crate harness for both AFL and libfuzzer fuzzer builds. 

This is done via 
1. For libfuzzer, we link it directly [here](https://github.com/dfinity/ic/blob/d8d7a18374c935cae8378afb26315be7517290df/bazel/fuzz_testing.bzl#L63)
2. For AFL, it's provided via [afl-clang-lto](https://github.com/dfinity/ic/blob/d8d7a18374c935cae8378afb26315be7517290df/bazel/conf/.bazelrc.build#L110) binary

Reply:
I ran the hourly pipeline on `eero/hermetic-toolchain-afl` and you could see the [failures](https://github.com/dfinity/ic/actions/runs/13651622698/job/38161290694#step:5:112) due to double linking

Reply:
Ah! I saw that on `-afl-zig` but didn't check here.

If I build the libfuzzer path without `link_libfuzzer`, there are all sorts of `undefined symbol` errors - I think due to a stdlib mismatch - and building/linking directly in the crate works around that.

I guess the direct links become noops because all the symbols are resolved by the earlier lib. I had these removed in an earlier draft, but brought them back because I thought maybe it had some effect on the `allow_main` targets.
Fuzzing is new to me - is `./bin/run-all-fuzzers.sh --libfuzzer` enough to see that things are working smoothly?

---

On the `-afl-zig` branch I split out `Cargo.Bazel.AFL.{json,toml}.lock` to handle the difference. Do you think that is reasonable here if it comes to it?

Reply:
Awesome! Thanks for the help, I've just pulled in these changes. I'll talk with @nmattia tomorrow before opening up the rest of the review 👀.

Reply:
We came up with another approach that avoids the extra lockfiles. The fuzzers look OK to me, but could you check that the `allow_main` variants are still fine? We'd still like to keep the `sandbox_libfuzzer` path around, right?

Too bad I can't get away with the +100000 diff 😅.

Reply:
Wow, this is awesome. I'm honestly surprised how the linker works fine for both the main / no_main variants. Everything looks good to me from fuzzing side 👍 

> We'd still like to keep the sandbox_libfuzzer path around, right?

Yep, while most of the moat is removed, we still need to run the binary under certain environment variables, so we would keep the `sandbox_libfuzzer` path around.

Thanks for this effort 🥇